### PR TITLE
Add RoCm scheduled CI & upgrade RoCm CI to PyTorch 2.1

### PR DESCRIPTION
**Pull request by fxmarty**
As per title

### CODE REVIEW

File: .github/workflows/self-scheduled-amd-caller.yml
Comment (fxmarty) [inline .github/workflows/self-scheduled-amd-caller.yml:14]: The `rocm-scheduled-ci` is temporary just in order to run on this PR: https://github.com/huggingface/transformers/actions/runs/6577295584

File: .github/workflows/self-push-amd-mi210-caller.yml
Comment (ydshieh) [reply .github/workflows/self-push-amd-mi210-caller.yml:21]: Yes, nice!

**Comment by ydshieh** [comment]
@fxmarty Could you verify that docker image could be built? For the AMD push CI, `docker/transformers-pytorch-amd-gpu` is not able to be built.

Note, `docker/transformers-pytorch-amd-gpu` is built on `[self-hosted, docker-gpu, amd-gpu, single-gpu, mi210]`, as the `ubuntu-latest` always gets not enough disk space error, no matter what I tried. I made it ran on `[self-hosted, docker-gpu, amd-gpu, single-gpu, mi210]` (maybe for a few days), but after that it gives permission error `EACCES: permission denied, mkdir '/home/github_actions/.docker/buildx/certs'`.

https://github.com/huggingface/transformers/actions/runs/6581781214

**Comment by fxmarty** [comment]
It appears the runners on MI210 are down.

File: .github/workflows/self-scheduled-amd.yml
Comment (ydshieh) [inline .github/workflows/self-scheduled-amd.yml:422]: Need to create a new channel and use it here before merge

File: .github/workflows/self-scheduled-amd.yml
Comment (ydshieh) [inline .github/workflows/self-scheduled-amd.yml:178]: From this job to `run_pipelines_torch_gpu` job , we need to have `if: always()`. Otherwise, if some jobs in previous stage fails, the current job won't be run, I believe:

> Use jobs.<job_id>.needs to identify any jobs that must complete successfully before this job will run. It can be a string or array of strings. If a job fails or is skipped, all jobs that need it are skipped unless the jobs use a conditional expression that causes the job to continue.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

File: .github/workflows/self-scheduled-amd.yml
Comment (ydshieh) [inline .github/workflows/self-scheduled-amd.yml:169]: The current Nvidia daily CI takes 17-20 hours. If `run_tests_multi_gpu` needs to wait `run_tests_single_gpu`, I am afraid the CI can't complete in 24 hours, despite `run_tests_single_gpu` use `max-parallel: 2`.

Do you remember how many runners are available for each flavor?

File: .github/workflows/self-scheduled-amd.yml
Comment (ydshieh) [inline .github/workflows/self-scheduled-amd.yml:436]: We might need to adjust the slack notification report to make it work due to the change in this PR: as it doesn't output the same number of test reports as the nvidia-one.

I can work on this before merge, but let me wait for the mi210 machine back to up.

**Review by ydshieh — COMMENTED** [review]
Thank you @fxmarty . Overall good for me, but left a few question/comments.

Something has to be change before merge (slack report script - I will take care of it).

**Comment by ydshieh** [comment]
Hi @mfuntowicz When you get some bandwidth, could you check how to bring back mi210 runners? Would be great if we can move toward the daily AMD CI (complete suite) and see the status :-)

File: .github/workflows/self-scheduled-amd-caller.yml
Comment (ydshieh) [reply .github/workflows/self-scheduled-amd-caller.yml:14]: Done

**Review by ydshieh — APPROVED** [review]
Thank you for this work @fxmarty @echarlaix !

It's ready, but we need a final 👍 from core maintainers, so let me request one review from @LysandreJik .

(Please ignore the conflict in `.github/workflows/build-docker-images.yml`)

Note: we still have the image build issue for which we need help from the infra team + the GPU visibility issue to be resolved.

**Review by LysandreJik — APPROVED** [review]
Yes, looks good to me! Clean work @fxmarty!

**Comment by fxmarty** [comment]
@ydshieh Probably did most of it! :smile:

**Comment by echarlaix** [comment]
thank you so much @ydshieh, you rock!

**Comment by LysandreJik** [comment]
You're right, thank you both :raised_hands: :hugs:

**Pull request merged**
